### PR TITLE
Add set-snapshot-address command 

### PR DESCRIPTION
**Pull request by thomaspanf**
wip

https://testnet.rocketpool.net/manage/snapshot-delegation

```
   set-snapshot-address, ssa    Set the address you want to use to represent your node on Snapshot
   clear-snapshot-address, csa  Clear the node's snapshot address
```

todo: 

- [x] gas estimations
- [x] unset snapshot address
- [ ] handle ENS names
- [x] handle revert messages

### CODE REVIEW

File: rocketpool-cli/pdao/commands.go [hunk 1]
```diff
@@ -79,6 +79,38 @@ func RegisterCommands(app *cli.App, name string, aliases []string) {
 				},
 			},
 
+			{
+				Name:      "set-snapshot-address",
+				Aliases:   []string{"ssa"},
+				Usage:     "Set the address you want to use when voting on Rocket Pool on-chain governance proposals, or the address you want to delegate your voting power to.",
+				UsageText: "rocketpool pdao set-voting-delegate address",
```
Comment (jshufro) [inline rocketpool-cli/pdao/commands.go:86]:
```suggestion
				UsageText: "rocketpool pdao set-snapshot-address address",
```

File: rocketpool/api/pdao/set-snapshot-address.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+	votingInitialized, err := network.GetVotingInitialized(rp, nodeAccount.Address, nil)
+	if !votingInitialized {
+		return nil, fmt.Errorf("Voting must be initialized to set a snapshot address. Use 'rocketpool pdao initialize-voting' to initialize voting first.")
+	}
+
+	// Slice signature string into v, r, s component of a signature giving node permission to use the given signer
+	str_v := signature[len(signature)-2:]
+	str_r := signature[:64]
+	str_s := signature[64:128]
```
Comment (jshufro) [inline rocketpool/api/pdao/set-snapshot-address.go:95]: can we add a function that expects a 129 byte 0x-prefixed EIP-712 signature and returns v/r/s as `v uint8` and `r, s []byte`?

And, ideally, a unit test :)

Comment (thomaspanf) [reply rocketpool/api/pdao/set-snapshot-address.go:95]: I truncated the 0x prefix [here](https://github.com/rocket-pool/smartnode/blob/54da653252683cf4b36bc804f993d69cc42a76ba/shared/utils/cli/validation.go#L85). Your suggestion is def a better way to handle it though. Will do, thanks for the review :)

**Comment by thomaspanf** [comment]
I marked this ready for review and switched the branch to `rocket-pool:set-snapshot-address` just in case Fornax wants to make edits while I'm away for a long weekend.

File: shared/utils/api/utils.go [hunk 3]
```diff
@@ -91,3 +94,41 @@ func IsTransactionDue(rp *rocketpool.RocketPool, startTime time.Time) (bool, tim
...
 
 }
+
+//  Expects a 129 byte 0x-prefixed EIP-712 signature and returns v/r/s as v uint8 and r, s *[32]byte
+
+func ParseEIP712(signature string) (uint8, *[32]byte, *[32]byte, error) {
+	if len(signature) != 132 || signature[:2] != "0x" {
+		return 0, nil, nil, fmt.Errorf("Invalid 129 byte 0x-prefixed EIP-712 signature while parsing: '%s'\n", signature)
+	}
+	signature = signature[2:]
```
Comment (jshufro) [inline shared/utils/api/utils.go:104]: definitely don't strip this outside the function that parses the signature.

File: shared/utils/api/utils.go [hunk 4]
```diff
@@ -91,3 +94,41 @@ func IsTransactionDue(rp *rocketpool.RocketPool, startTime time.Time) (bool, tim
 	return isDue, timeUntilDue, nil
 
 }
+
+//  Expects a 129 byte 0x-prefixed EIP-712 signature and returns v/r/s as v uint8 and r, s *[32]byte
+
+func ParseEIP712(signature string) (uint8, *[32]byte, *[32]byte, error) {
```
Comment (jshufro) [inline shared/utils/api/utils.go:100]: can we return a pointer to a struct with fields for v, r, s instead of returning a single scalar and 3 pointers

**Pull request merged**